apps/server/src/modules/learnroom/uc/dashboard.uc.spec.ts

```diff
@@ -56,11 +78,40 @@ describe('dashboard uc', () => {
 				const dashboard = new DashboardEntity('someid', { grid: [], userId });
 				return Promise.resolve(dashboard);
 			});
+			jest.spyOn(courseRepo, 'findAllByUserId').mockImplementation(() => {
+				return Promise.resolve([[], 0]);
+			});
 			const dashboard = await service.getUsersDashboard('userId');
 
 			expect(dashboard instanceof DashboardEntity).toEqual(true);
 			expect(spy).toHaveBeenCalledWith('userId');
 		});
+
+		it('should synchronize which courses are on the board', async () => {
+			const userId = 'userId';
+			const dashboard = new DashboardEntity('someid', { grid: [], userId });
+			const dashboardRepoSpy = jest.spyOn(repo, 'getUsersDashboard').mockImplementation((id: EntityId) => {
+				return Promise.resolve(dashboard);
+			});
+			const courses = new Array(5).map(() => ({} as Course));
+			const courseRepoSpy = jest.spyOn(courseRepo, 'findAllByUserId').mockImplementation((id, filters?, options?) => {
+				return Promise.resolve([courses, 5]);
+			});
+			const syncSpy = jest.spyOn(dashboard, 'setLearnRooms');
+			const persistSpy = jest.spyOn(repo, 'persistAndFlush');
+
+			const result = await service.getUsersDashboard('userId');
+
+			expect(result instanceof DashboardEntity).toEqual(true);
+			expect(dashboardRepoSpy).toHaveBeenCalledWith('userId');
+			expect(courseRepoSpy).toHaveBeenCalledWith(
+				userId,
+				{ onlyActiveCourses: true },
```

Comment:
I sadly guess this needs to be changed, right? ^^

Reply:
actually not. This is for the dashboard view, where we only want to add active courses, which is the difference to the list where all courses (including archived) are shown.